File: app/src/main/java/lt/markmerkk/DBProdExecutor.java
package lt.markmerkk;

import java.net.URI;
import java.net.URISyntaxException;
import java.nio.file.Paths;
import javax.annotation.PostConstruct;
import javax.annotation.PreDestroy;
import javax.inject.Inject;
import lt.markmerkk.entities.database.DBBaseExecutor;
import lt.markmerkk.utils.UserSettingsImpl;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by mariusmerkevicius on 11/22/15.
 */
public class DBProdExecutor extends DBBaseExecutor {
  public static final Logger logger = LoggerFactory.getLogger(DBProdExecutor.class);

  @Inject
  UserSettingsImpl settings;

  @PostConstruct
  private void initialize() {
    logger.debug("Applying database migrations");
    if (settings.getVersion() <= 0) {
      logger.debug("Found unversioned database! Flushing database with new version.");
      settings.setVersion(Main.VERSION_CODE);
    }
    migrate();
  }

  @PreDestroy
  private void destroy() {
  }

  @Override
  protected String database() {
    return Main.CFG_PATH + "wt4_1.db";
  }

  @Override
  protected URI migrationScriptPath() {
    try {
      return getClass().getResource("/changelog_1.xml").toURI();
    } catch (URISyntaxException e) {
      return null;
    }
  }

  @Override
  protected URI migrationExportPath() {
    return Paths.get(Main.CFG_PATH).toUri();
  }

}

File: app/src/main/java/lt/markmerkk/DBProdExecutor.kt
package lt.markmerkk

import lt.markmerkk.entities.SimpleLog
import lt.markmerkk.entities.database.DBBaseExecutor
import lt.markmerkk.entities.jobs.CreateJobIfNeeded
import org.slf4j.LoggerFactory
import java.net.URI
import java.net.URISyntaxException
import java.nio.file.Paths

/**
 * Created by mariusmerkevicius on 11/22/15.
 */
class DBProdExecutor(
        private val config: Config,
        private val settings: UserSettings
) : DBBaseExecutor() {

    init {
//        logger.debug("Applying database migrations")
//        if (settings.version <= config.versionCode) {
//            logger.debug("Database older version, adding migrations.")
//            settings.version = config.versionCode
//            migrate()
//        }
        executeOrThrow(CreateJobIfNeeded(SimpleLog::class.java))
    }

    override fun database(): String {
        return config.cfgPath + "wt4_1.db"
    }

    override fun migrationScriptPath(): URI {
        try {
            return javaClass.getResource("/changelog_1.xml").toURI()
        } catch (e: URISyntaxException) {
            throw IllegalStateException("Can't find migration config path")
        }
    }

    override fun migrationExportPath(): URI {
        return Paths.get(config.cfgPath).toUri()
    }

    companion object {
        private val logger = LoggerFactory.getLogger(DBProdExecutor::class.java)!!
    }

}
